chia/wallet/wallet_state_manager.py

```diff
@@ -1419,6 +1419,7 @@ async def _add_coin_states(
                                 fee = 0
 
                                 to_puzzle_hash = None
+                                coin_spend = await fetch_coin_spend_for_coin_state(coin_state, peer)
```

Comment:
shouldn't this be under `if derivation_record is None` ?

Reply:
We just need to fetch the parent coin spend once. for coin in additions is child coin level.

Reply:
No, for here we are already sure we know the coin puzzle hash so derivation_record cannot be none.